graphbolt/src/concurrent_id_hash_map.h

```diff
@@ -158,6 +158,16 @@ class ConcurrentIdHashMap {
    */
   inline void InsertAndSet(IdType key, IdType value);
 
+  /**
+   * @brief Insert a key into the hash map. If the key exists, set the value
+   * with the smaller value.
+   *
+   * @param id The key to be inserted.
+   * @param value The value to be set for the `key`.
+   *
+   */
+  inline void InsertAndSetSmaller(IdType id, IdType value);
```

Comment:
Do we still need InsertAndSet after this work? If not, why not user replace?

Reply:
It's different from `InsertAndSet` so I created another function. Now `InsertAndSet` is not used. We can delete it or save it for possible future use because it's a common operation for a hash map.